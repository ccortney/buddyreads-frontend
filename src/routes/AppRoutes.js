import {Route, Routes, Navigate} from "react-router-dom";
import Homepage from "../homepage/Homepage";
import LoginForm from "../forms/LoginForm";
import SignupForm from "../forms/SignupForm"
import Dashboard from  "../dashboard/Dashboard";
import ProfileForm from "../forms/ProfileForm";
import PrivateRoute from "./PrivateRoute"
import BuddyReadDetails from "../buddyread/BuddyReadDetails";

/** Site-wide routes.
 *
 * Parts of site should only be visitable when logged in. Those routes are
 * wrapped by <PrivateRoute>, which is an authorization component.
 *
 * Visiting a non-existant route redirects to /.
 */

const AppRoutes = ({login, signup, editProfile}) => {

    return (
        <Routes>
            <Route path="/" element={<Homepage/>}/>
            <Route path="/login" element={<LoginForm login={login}/>}/>
            <Route path="/signup" element={<SignupForm signup={signup}/>}/>
            <Route path="/profile" element={<PrivateRoute/>}>
                <Route path="/profile" element={<ProfileForm editProfile={editProfile}/>}/>
            </Route>
            <Route path="/dashboard" element={<PrivateRoute/>}>
                <Route path="/dashboard" element={<Dashboard />}/>
            </Route>
            <Route path="/buddyreads/:id" element={<PrivateRoute/>}>
                <Route path="/buddyreads/:id" element={<BuddyReadDetails />}/>
            </Route>
            <Route path="*" element={<Navigate to="/"/>}/>
        </Routes>
    )
} 

export default AppRoutes;