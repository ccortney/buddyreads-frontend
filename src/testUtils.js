import UserContext from "./auth/UserContext";

const demoUser = {
  id: 0,
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
