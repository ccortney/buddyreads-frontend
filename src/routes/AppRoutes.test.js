import { render } from "@testing-library/react";
import AppRoutes from "./AppRoutes";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <AppRoutes />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <AppRoutes />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
