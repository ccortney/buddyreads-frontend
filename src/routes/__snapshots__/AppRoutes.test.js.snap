// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-iw00ie-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-10fld7o-MuiTypography-root"
      >
        BUDDY READS
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Choose a book. Find a friend. Discuss as you go.
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Welcome back,  !
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
